### ISSUE TITLE
[BaggagePropagator] Do not encode and decode keys

### ISSUE THREAD

**Issue by pellared** [open-telemetry/opentelemetry-dotnet#5479]
According to the spec only the value should be encoded. 

The key is not expected to be percent encoded. More:
- https://www.w3.org/TR/baggage/#key
- https://github.com/w3c/baggage/issues/131
- https://github.com/open-telemetry/opentelemetry-go/issues/4946

If the key is not W3C Baggage spec compliant then key-value pair must not be propagated and a warning should be emitted.

**Comment by martincostello** [comment]
@nabutabu Are any of the PRs you've had merged or in-flight going to close this?

**Comment by nabutabu** [comment]
> [@nabutabu](https://github.com/nabutabu) Are any of the PRs you've had merged or in-flight going to close this?

Yes #7051 fixes this. This issue is a child issue of https://github.com/open-telemetry/opentelemetry-dotnet/issues/5689
